include/prometheus/family.h

```diff
@@ -67,14 +67,26 @@ T& Family<T>::Add(const std::map<std::string, std::string>& labels,
   }
 #endif
 
-  std::lock_guard<std::mutex> lock{mutex_};
   auto hash = hash_labels(labels);
-  auto metric = new T(std::forward<Args>(args)...);
+  std::lock_guard<std::mutex> lock{mutex_};
+  auto metrics_iter = metrics_.find(hash);
+
+  if (metrics_iter != metrics_.end()) {
+#ifndef NDEBUG
+    auto labels_iter = labels_.find(hash);
+    assert(labels_iter != labels_.end());
+    const auto &old_labels = labels_iter->second;
+    assert(labels == old_labels);
+#endif
+    return *metrics_iter->second;
+  } else {
```

Comment:
Minor nitpick: you could leave out the else because you already returned at the end of the last block